### PR TITLE
feat: Per-request Key authorization #124

### PR DESCRIPTION
**Pull request by astsiapanay**

### CODE REVIEW

File: src/main/java/com/epam/aidial/core/config/FileConfigStore.java [hunk 1]
```diff
@@ -24,7 +23,7 @@ public final class FileConfigStore implements ConfigStore {
 
     private final String[] paths;
     private volatile Config config;
-    private final Map<String, String> deploymentKeys = new HashMap<>();
+    private final Map<String, ApiKeyData> keys = new HashMap<>();
```
Comment (artsiomkorzun) [inline src/main/java/com/epam/aidial/core/config/FileConfigStore.java]: Please move it to a different class. Config store just loads a config. Config should be immutable. Managing keys is a different responsibility and should be done outside.

Comment (astsiapanay) [reply src/main/java/com/epam/aidial/core/config/FileConfigStore.java]: fixed

File: src/main/java/com/epam/aidial/core/config/FileConfigStore.java [hunk 2]
```diff
@@ -83,21 +80,20 @@ private void load(boolean fail) {
...
                 Application application = entry.getValue();
                 application.setName(name);
-                associateDeploymentWithApiKey(config, application);
             }
 
             for (Map.Entry<String, Key> entry : config.getKeys().entrySet()) {
                 String key = entry.getKey();
                 Key value = entry.getValue();
                 value.setKey(key);
+                apiKeyStore.addProjectKey(value);
```
Comment (artsiomkorzun) [inline src/main/java/com/epam/aidial/core/config/FileConfigStore.java]: When do you remove? e.g. a config changes and does not have old keys.

Comment (astsiapanay) [reply src/main/java/com/epam/aidial/core/config/FileConfigStore.java]: good catch. I'll remove all project keys before adding new ones from the updated config.

Comment (astsiapanay) [reply src/main/java/com/epam/aidial/core/config/FileConfigStore.java]: fixed

**Review by artsiomkorzun — DISMISSED** [review]

**Comment by astsiapanay** [comment]
/deploy-review
>Deployment status: success

**Comment by astsiapanay** [comment]
/deploy-review
>Deployment status: success

**Comment by astsiapanay** [comment]
/deploy-review
>Deployment status: success

**Comment by astsiapanay** [comment]
/deploy-review
>Deployment status: success

**Comment by astsiapanay** [comment]
/deploy-review
>Deployment status: success

**Comment by astsiapanay** [comment]
/deploy-review
>Deployment status: success

**Review by artsiomkorzun — APPROVED** [review]

**Pull request merged**